src/NLog/Config/PropertyTypeConverter.cs

```diff
@@ -43,28 +44,35 @@ class PropertyTypeConverter : IPropertyTypeConverter
         /// <inheritdoc/>
         public object Convert(object propertyValue, Type propertyType, string format, IFormatProvider formatProvider)
         {
-            if (propertyType == null || propertyValue == null || propertyValue.GetType() == propertyType || propertyType == typeof(object))
+            if (!NeedToConvert(propertyValue, propertyType))
             {
                 return propertyValue;
             }
 
+            var nullableType = Nullable.GetUnderlyingType(propertyType);
+            var type = nullableType ?? propertyType;
             if (propertyValue is string propertyString)
             {
                 propertyValue = propertyString = propertyString.Trim();
 
-                if (propertyType == typeof(DateTime))
+                if (nullableType != null && StringHelpers.IsNullOrWhiteSpace(propertyString))
+                {
+                    return null;
+                }
+                
+                if (type == typeof(DateTime))
```

Comment:
Could consider having a conversion-dictionary. Where common types are registered along with their converter. Similar to what has been implemented in PropertyHelper. The conversion dictionary could have the type `DateTime`, `DateTime?`, `Guid`, `Guid?`, `int`, `int?`, `bool`, `bool?`, `string`. This would make the conversion path quicker.

Reply:
I was think the latter, as one didn't have to lookup the underlying for common types, as one would just register both-types in the lookup-dictionary. Just a thought.

Reply:
Maybe nice to do this in another PR :)